### ISSUE TITLE
Indexing C4 no clean collection

### ISSUE THREAD

**Issue by MarcosFP97** [castorini/anserini#1594]
Hi,

In a previous [issue](https://github.com/castorini/anserini/issues/1579), I asked about the difference between `Collection` and `Generator` classes, and the C4 collection example raised up.

The thing is that I am working with the no clean version of this collection under the TREC 2021 Health Misinformation Track, and I am trying to index it using the custom classes provided by Anserini:

```
nohup sh target/appassembler/bin/IndexCollection -collection C4Collection -input ~/c4/en.noclean.withdocnos -index indexes/C4 -generator C4Generator -threads 44 -storePositions -storeDocvectors -storeRaw >& logs/log.C4 &
```

The problem is that the generator is considering the whole JSON as the raw field of the document, and it is setting the incorrect docId. Instead, I would like the generator to grab each field (docno, text, timestamp, etc.) and promulgate them to the index.

To that end, I am not sure if I will need to create my custom index, as explained [here](https://github.com/castorini/pyserini#how-do-i-index-and-search-my-own-documents).

Thanks in advance!

**Comment by MarcosFP97** [comment]
[EDIT] I think that the problem is in class C4Collection.java since it generates the docId from the path instead of reading the JSON field. Moreover, I also think it is not reading correctly the JSON fields.

I will do a PR with the new collection class implemented, but I am open to suggestions.

**Comment by lintool** [comment]
Yes, you'll likely need to implement another Collection class. The currently implemented classes have associated test cases, which you can use as a reference for a new implementation.

**Comment by MarcosFP97** [comment]
Perfect, I will implement it and do the PR ASAP :)

**Comment by lintool** [comment]
Great, thanks - looking forward!

**Comment by MarcosFP97** [comment]
Here it is my [PR](https://github.com/castorini/anserini/pull/1596) :)

**Comment by lintool** [comment]
Closed by #1596
